src/jit/compiler.h

```diff
@@ -661,23 +701,23 @@ class  LclVarDsc
     bool                IsFloatRegType() const
                         {
                             return
-#ifdef _TARGET_ARM_
+#ifdef FEATURE_HFA
                                 lvIsHfaRegArg ||
 #endif
                                 isFloatRegType(lvType);
                         }
-#ifdef _TARGET_ARM_
+#ifdef FEATURE_HFA
     var_types           GetHfaType() const
                         {
-                            assert(lvIsHfaRegArg);
-                            return lvIsHfaRegArg ? (lvHfaTypeIsFloat ? TYP_FLOAT : TYP_DOUBLE) : TYP_UNDEF;
+                            assert(lvIsHfa());
+                            return lvIsHfa() ? (lvHfaTypeIsFloat ? TYP_FLOAT : TYP_DOUBLE) : TYP_UNDEF;
                         }
     void                SetHfaType(var_types type)
                         {
                             assert(varTypeIsFloating(type));
                             lvHfaTypeIsFloat = (type == TYP_FLOAT);
```

Comment:
In view of preferring #ifdef'ed code inside the method rather than outside it, implementation of this method could be put inside #ifdef FEATURE_HFA rather than the entire method.  That would allow the callers of this method not to require to enclose in a #ifdef.

.e.g.
```
void setHfaType(var_types type)
{
#ifdef FEATURE_HFA
    assert(varTypeIsFloating(type));
    lvHfaTypeIsFloat = (type == TYP_FLOAT);
#endif 
}
```